src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,127 @@
+        }
+        this.saveFile = savedTaskFile;
+    }
+
+    public ArrayList<Task> load() {
+        ArrayList<Task> userTasks = new ArrayList<>();
+        BufferedReader reader;
+        try {
+            reader = new BufferedReader(new FileReader(saveFile, Charset.defaultCharset()));
+            String taskStr = reader.readLine();
+            while (taskStr != null) {
+                String[] parts = taskStr.split("\\|");
+                boolean completed = parts[1].equals("1");
+                String taskDescription = parts[2];
+                if (parts.length == 3) {
+                    Todo newTodo = new Todo(taskDescription, completed);
+                    userTasks.add(newTodo);
+                }
+                if (parts.length == 4) {
+                    try {
+                        String[] deadlineParts = parts[3].split(" ");
+                        String deadlineDate = deadlineParts[0];
+                        String deadlineTime = deadlineParts[1];
+                        LocalDate parsedDate = LocalDate.parse(deadlineDate);
+                        Date parsedTime = new SimpleDateFormat("hh:mm").parse(deadlineTime);
+                        Deadline newDeadline = new Deadline(taskDescription, parsedDate, parsedTime, completed);
+                        userTasks.add(newDeadline);
+                    } catch (DateTimeParseException | ParseException e) {
+
+                    }
+                }
+                if (parts.length == 5) {
+                    try {
+                        String[] eventStartParts = parts[3].split(" ");
+                        String[] eventEndParts = parts[4].split(" ");
+                        String eventStartDate = eventStartParts[0];
+                        String eventStartTime = eventStartParts[1];
+                        String eventEndDate = eventEndParts[0];
+                        String eventEndTime = eventEndParts[1];
+                        LocalDate parsedStartDate = LocalDate.parse(eventStartDate);
+                        LocalDate parsedEndDate = LocalDate.parse(eventEndDate);
+                        Date parsedStartTime = new SimpleDateFormat("hh:mm").parse(eventStartTime);
+                        Date parsedEndTime = new SimpleDateFormat("hh:mm").parse(eventEndTime);
+                        Event newEvent = new Event(taskDescription, parsedStartDate, parsedStartTime,
+                                parsedEndDate, parsedEndTime, completed);
+                        userTasks.add(newEvent);
+                    } catch(DateTimeParseException | ParseException e) {
+
+                    }
+                }
+                taskStr = reader.readLine();
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return userTasks;
+    }
+
+    public static void saveTasksToFile(ArrayList<Task> taskList) {
+        try {
+            String toWrite = "";
```

Comment:
A little pedantic, but it might be better to use a noun for this String, such as tasksToWrite